sound/soc/sof/intel/hda-loader.c

```diff
@@ -87,6 +87,7 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
```

Comment:
@ranj063 @plbossart FWIW, I'm tempted to propose removing all boot_iteration based logic. It seems we are creating just too much logic just to hide failures on ICL (https://github.com/thesofproject/sof/issues/3395). If this targets 5.11, we have a chance to get the ICCMAX change in and that may potentially fix this problem for ICL. If yes, we could revert to simpler times when a failure to boot or init is simply just a failure.

Reply:
@kv2019i that can be done as a separate PR no? This change is required for the error we're seeing on some devices where the FW boot fails at the cl_cop_fw() stage and we have no ROm status dumps

Reply:
@ranj063 Do we need this merged in sof-dev (i.e. gets better logs from our CI) or deployed to upstream? I'm ok to push this in for former if we agree we wait until ICCMAX ICL implementation and we can see whether we can drop the boot-iteration log-level logic altogther for 5.11. 

@fredoh9 Do we have some draft version of ICCMAX driver+fw? It would be interesting to see whether it helps with sof#3395 or not.

Reply:
@kv2019i this is needed for upstream

Reply:
@ranj063 So then if we can potential intercept ICCMAX change for 5.11, then we could drop this in upstream as well and go back to old behaviour of printing errors whenever the boot fails. If this is rare enough, it's ok to print as errors. If the ICL ICCMAX won't help and/or no line of sight to getting it in for 5.11, then I'm ok to go with this.

Reply:
@kv2019i no one knows is ICCMAX will help prevent multiple boot attempts on ICL.

Reply:
@ranj063 Ack, but we are not anyways getting these upstream until 5.11, so we have some time to get data on that. That's why I was asking for @fredoh9 if we have a draft version ready. The problem is systematic on ICL now, so a few test runs on CI with the draft patch in, could help to give some data whether this helps at all. If I'm just too optimistic on the ICCMAX, then ok, let's put this in.